MdeModulePkg/Core/Dxe/Mem/HeapGuard.c

```diff
@@ -252,6 +254,14 @@ FindGuardedMemoryMap (
                  );
       ASSERT_EFI_ERROR (Status);

```

Comment:
Not entirely related to these changes, but this is the kind of place something other than just asserting should be done if the allocation fails.

Reply:
Converted this to a REBASE && FF PR. The second commit updates this section and another similar section to branch execution if the allocation fails in addition to ASSERTing.